osg/short-term.md

```diff
@@ -0,0 +1,292 @@
+more detail later.
+
+### stdout
+
+This setting contains the name of the file that should contain the redirected standard output produced by the tool.
+
+### stderr
+
+This setting contains the name of the file that should contain the redirected standard error output produced by the
+tool.
+
+## Initializing the iRODS Connection
+
+Initializing the iRODS connection is actually fairly simple if you're using version 4.x of the icommands, which is
+highly recommended. The username in the `irods_user` configuration setting is guaranteed to have an empty password, so
+iRODS can be initialized by creating a file called `$HOME/.irods/irods_environment.json`. This file should be in the
+following format:
+
+```
+{
+    "irods_user_name": "<username>",
+    "irods_host": "<host-name>",
+    "irods_port": <port-number>,
+    "irods_zone_name": ""
+}
+```
+
+For the time being, the zone name should always be equal to the empty string. The rest of the settings are extracted
+from the job configuration file. After creating this file, `iget` and `iput` can be used in conjunction with the iRODS
+tickets to retrieve files from or push files to iRODS.
+
+If you're using custom iRODS credentials with a password then (assuming you're not using Jargon) `iinit` must be called
+before using any of the iRODS icommands. `iinit` only runs in interactive mode, so it will be necessary to send the
+password to `iinit` using redirection. Using custom iRODS credentials is expected to be rare, so precise instructions
+for running `iinit` are beyond the scope of this documentation. Similarly, using Jargon is beyond the scope of this
+documentation.
+
+## Sending Job Status Updates
+
+Job status updateS can and should be sent to the DE using HTTP POST requests. The complete URL used to send job status
+updates for the current job is provided in the job configuration file. The request body is a JSON object consisting of
+three fields. The `state` field contains the current job status. The `message` field contains a brief description of
+what the job is currently doing. The `hostname` field contains the host name or IP address tht the request is being sent
+from.
+
+### Status Codes
+
+There are three job status codes that can be used in the wrapper script: `running`, `completed` and `failed`. A status
+update with the status code `running` should be sent immediately after the job configuration file is parsed. This
+informs the user that the job has started. The wrapper script may send as many `running` job status updates as it
+wants. It is often helpful to send job status updates when the wrapper script is about to start downloading files, about
+to kick off the analysis itself, and about to start uploading output files.
+
+The other two status codes, `completed` and `failed`, are intended to be sent only once just before the wrapper script
+exits. If the job completed successfully then the status code should be `completed`. Otherwise, it should be `failed`.
```

Comment:
We should probably specifically mention that care should be taken to always send at least one of these codes -- unless we need to provide other guidance to deal with the fact OSG can pre-empt jobs and reschedule them? I'm not totally sure what we can do to detect that case, though.